packages/@webex/plugin-cc/src/AgentProfile/AgentProfile.ts

```diff
@@ -0,0 +1,108 @@
+import {AgentProfileResponse} from './types';
+import AgentProfileService from '../AgentProfileService/AgentProfileService';
+import {
+  DesktopProfileResponse,
+  ListAuxCodesResponse,
+  ListTeamsResponse,
+  UserResponse,
+} from '../AgentProfileService/types';
+import {WebexSDK} from '../types';
+import {attributes, page, pageSize} from '../AgentProfileService/constants';
+
+export default class AgentProfile {
+  ciUserId: string;
+  agentProfile: AgentProfileResponse;
+  webex: WebexSDK;
+  wccAPIURL: string;
+
+  constructor(ciUserId: string, webex: WebexSDK, wccAPIURL: string) {
+    this.webex = webex;
+    this.ciUserId = ciUserId;
+    this.agentProfile = {
+      userDetails: {} as UserResponse['body'],
+      agentDesktopProfile: {} as DesktopProfileResponse['body'],
+      teamsList: [] as ListTeamsResponse['body'],
+      auxCodesList: [] as ListAuxCodesResponse['body']['data'],
+    };
+    this.wccAPIURL = wccAPIURL;
+  }
+
+  /**
+   * Method to get Agent Profile by providing ciUserId.
+   * @param {string} ciUserId The CI ID of a User.
+   * @returns {Promise<AgentProfileResponse>} A promise that eventually resolves to an API response and return configuration of an Agent.
+   */
+
+  public async getAgentProfile(ciUserId: string): Promise<AgentProfileResponse> {
+    try {
+      const orgId = await this.webex.credentials.getOrgId();
+      if (!ciUserId || !orgId) {
```

Comment:
Is there a chance of these 2 fields being null or empty?

Reply:
ciUserId we are getting from Mercury connection, so it might be the case where the ciUserId is invalid and orgId we are fetching from webex object using getOrgId function, if something wrong happens from webex side we might get invalid orgId. So, I tried to not hit the API server with invalid data and tried to reject the request before itself.

Reply:
Could we find out when the ciUserId comes as an invalid value?